zerver/actions/streams.py

```diff
@@ -315,7 +315,7 @@ def do_unarchive_stream(stream: Stream, new_name: str, *, acting_user: UserProfi
         internal_send_stream_message(
             sender,
             stream,
-            channel_events_topic_name(),
+            channel_events_topic_name(stream),
```

Comment:
This change in calling convention could ideally be in the prep commit -- you can have a function with extra parameters that it ignores. I think with https://github.com/zulip/zulip/pull/34961/commits/45e1ad95242a89eafaf2a2710c02a789b4ec3c9a#r2188676265, you'll also want to include the `realm` as a parameter.